src/apps/appstore/app.js

```diff
@@ -393,8 +471,296 @@ define(function(require) {
+					});
+			};
+
+			$('.market-settings #save').on('click', function(e) {
+				e.preventDefault();
+				const saveThis = $(this);
+				saveThis.prop({ disabled: true });
+
+				if (monster.ui.valid(configForm)) {
+					const formData = monster.ui.getFormData('configuration_form');
+					marketAction({ action: 'api_url', api_url: formData.api_url }, 'api_url', saveThis);
+				} else {
+					saveThis.prop({ disabled: false });
+				}
+			});
+
+			$('.market-settings #disable').on('click', function(e) {
+				e.preventDefault();
+				const saveThis = $(this);
+
+				self.confirmDisableDialog(function() {
+					saveThis.prop({ disabled: true });
+					marketAction({ action: 'disable' }, 'enabled', saveThis);
+				});
+			});
+
+			$('.market-settings #disconnect').on('click', function(e) {
+				e.preventDefault();
+				const saveThis = $(this);
+
+				self.confirmDisconnectDialog(function() {
+					saveThis.prop({ disabled: true });
+					marketAction({ action: 'unlink' }, 'unlink', saveThis);
+				});
+			});
+		},
+
+		confirmDisableDialog: function(callbackSuccess) {
+			const self = this;
+			const content = self.i18n.active().marketplace.dangerDisableDialog.content;
+			monster.ui.confirm(
+				content,
+				function() {
+					callbackSuccess && callbackSuccess();
+				}
+			);
+		},
+
+		confirmDisconnectDialog: function(callbackSuccess) {
+			const self = this;
+			const actionKey = self.i18n.active().marketplace.dangerDisconnectDialog.actionKey;
+			const confirmPopup = monster.ui.confirm(
+				$(self.getTemplate({
+					name: 'dangerConfirmDialog',
+					data: {
+						clusterId: self.appstoreData.marketplace.cluster_id
+					}
+				})),
+				function() {
+					callbackSuccess && callbackSuccess();
+				},
+				null,
+				{
+					title: self.i18n.active().marketplace.dangerDisconnectDialog.title,
+					confirmButtonText: self.i18n.active().marketplace.dangerDisconnectDialog.action_btn,
+					htmlContent: true
+				}
+			);
+
+			confirmPopup.find('#confirm_button').prop('disabled', true);
+
+			confirmPopup.find('#action_input').on('keyup', function() {
+				if ($(this).val() === actionKey) {
+					confirmPopup.find('#confirm_button').prop('disabled', false);
+				} else {
+					confirmPopup.find('#confirm_button').prop('disabled', true);
+				}
+			});
+		},
+
+		renderMarketWelcome: function(parent, appstoreData) {
+			const self = this;
+			const template = $(self.getTemplate({
+				name: 'marketWelcome'
+			}));
+
+			parent
+				.find('.right-container')
+				.fadeOut(function() {
+					$(this)
+						.empty()
+						.append(template)
+						.fadeIn();
+					self.bindMarketWelcomeEvents(parent);
+				});
+		},
+
+		bindMarketWelcomeEvents: function(parent) {
+			const self = this;
+			const maybeSetApiUrl = function() {
+				if (!self.appstoreData.marketplace.api_url && monster.config.api.default) {
+					self.updateMarketConnector({
+						action: 'api_url',
+						api_url: monster.config.api.default
+					});
+				}
+			};
```

Comment:
Looks like this logic is used in more than one place, os it might be a good idea to extract it in its own method.